steps.go

```diff
@@ -697,26 +700,38 @@ func (tc *TestContext) cPUMetricsOfAllInstancesInTheInstancePoolMustBeVisibleInP
 			tc.logger.Debugf("failed to get instance pool (%v)", data.Data.ResultType)
 			continue
 		}
+
+		var instancePoolIps []string
+		var metricIpsSliced []string
 		metricIps := map[string]bool{}
 		for _, record := range data.Data.Result {
 			parts := strings.SplitN(record.Metric.Instance, ":", 2)
 			metricIps[parts[0]] = true
 		}
+		for k := range metricIps {
+			metricIpsSliced = append(metricIpsSliced, k)
+		}
 		allIpsFound := true
 		for _, vm := range instancePool.VirtualMachines {
 			for _, nic := range vm.Nic {
-				if nic.IsDefault {
+				if nic.IsDefault && nic.IPAddress != nil {
+					instancePoolIps = append(instancePoolIps, nic.IPAddress.String())
 					if _, ok := metricIps[nic.IPAddress.String()]; !ok {
 						allIpsFound = false
 					}
 				}
 			}
 		}
+
+		tc.logger.Debugf("expecting following targets (vms in the instancepool): %v", instancePoolIps)
+		tc.logger.Debugf("found the following targets (monitored by Prometheus): %v", metricIpsSliced)
```

Comment:
### Reason
Logging of actual `instancePoolIps` vs those present in the current Prometheus config (similar to the found vs expected target logging for previous service discovery).